deno/payloads/v8/channel.ts

```diff
@@ -32,35 +32,87 @@ export interface APIPartialChannel {
 }
 
 /**
- * https://discord.com/developers/docs/resources/channel#channel-object-channel-structure
+ * This interface is used to allow easy extension for other channel types. While
+ * also allowing `APIPartialChannel` to be used without breaking.
  */
-export interface APIChannel extends APIPartialChannel {
+export interface APIChannelBase<T extends ChannelType> extends APIPartialChannel {
+	type: T;
+}
+
+export type TextBasedChannels = ChannelType.DM | ChannelType.GroupDM | ChannelType.GuildNews | ChannelType.GuildText;
```

Comment:
Is this really needed? Can it be inlined instead? If it is needed, please rename it to `TextChannelTypes`

Reply:
Wouldn’t that be confusing though since Text Channel is a type of channel?

Reply:
It's GuildTextChannel, so.. not really, no

Reply:
Because this isn't discord.js? This is a type that represents all the possible channels that may have text in them, so its a `TextChannelTypes` union 🤷‍♂️

Reply:
Yeah I know it isn't, I was just comparing it to d.js since it can cause confusion with the name used there, but you're the boss 👍